### PR TITLE
fix(Core/Mailhandler):  fix in premium module

### PR DESCRIPTION
**Pull request by pangolp**
##### CHANGES PROPOSED:
-  It does not work the sending of mails by means of the module.

##### ISSUES ADDRESSED:
- Closes

##### TESTS PERFORMED:
- Tests carried out in Windows 10. Using the premium azeroth module.

##### Target branch(es):
- [x] Master

 

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/game/Handlers/MailHandler.cpp [hunk 1]
```diff
@@ -22,8 +22,11 @@ bool WorldSession::CanOpenMailBox(uint64 guid)
 {
     if (guid == _player->GetGUID())
     {
-        sLog->outError("%s attempt open mailbox in cheating way.", _player->GetName().c_str());
-        return false;
+        return true;
```
Comment (FrancescoBorzi) [inline src/server/game/Handlers/MailHandler.cpp]: why changing this return?

Comment (Viste) [reply src/server/game/Handlers/MailHandler.cpp]: to allow open mailbox in cheating way i guess

Comment (Bogir) [reply src/server/game/Handlers/MailHandler.cpp]: it must be something like:

```suggestion
if (guid == _player->GetGUID())
{
  QueryResult result_prem = CharacterDatabase.PQuery("SELECT * FROM premium WHERE active = 1 AND AccountId = %u", _player->GetSession()->GetAccountId());
  if (!result_prem) {
    sLog->outError("%s attempt open mailbox in cheating way.", _player->GetName().c_str());
    return false;
  } else { 
    return true;
}
```

Comment (Bogir) [reply src/server/game/Handlers/MailHandler.cpp]: What is the premium-module you are talking about?

**Comment by pangolp** [comment]
Yeah, it's the premium module. After I make the pull request, I realize that maybe it wasn't a theme of the core, nose of the module as you mention. It was a topic of conversation that you had inside the module, where you proposed that change. 
**https://github.com/azerothcore/mod-premium/issues/8**

**Pull request closed without merging**